pkg/apis/operator/v1alpha1/tektonconfig_defaults.go

```diff
@@ -29,9 +31,23 @@ func (tc *TektonConfig) SetDefaults(ctx context.Context) {
 	tc.Spec.Trigger.setDefaults()
 
 	if IsOpenShiftPlatform() {
+		if tc.Spec.Platforms.OpenShift.PipelinesAsCode == nil {
+			tc.Spec.Platforms.OpenShift.PipelinesAsCode = &PipelinesAsCode{
+				Enable: ptr.Bool(true),
+			}
+		} else {
+			tc.Spec.Addon.EnablePAC = nil
```

Comment:
for an upgrade, should not we need to do this every time?

Reply:
it is done everytime, except the first.
also in
```
func (ta *TektonAddon) SetDefaults(ctx context.Context) {
```